docs/2.guide/3.going-further/12.custom-router.md

```diff
@@ -0,0 +1,144 @@
+---
+title: "Custom router"
+description: "In Nuxt 3, your routing is defined by the structure of your files inside the pages directory. However, since it uses vue-router under the hood, Nuxt offers you several ways to add custom routes in your project."
+---
+
+## Adding custom routes
+
+In Nuxt 3, your routing is defined by the structure of your files inside the [pages directory](/docs/guide/directory-structure/pages). However, since it uses [vue-router](https://router.vuejs.org/) under the hood, Nuxt offers you several ways to add custom routes in your project.
+
+### Using router config
+
+Using [router options](/docs/2.guide/3.going-further/12.custom-router#router-options), you can optionally override routes using a function that accepts scanned routes and returns customized routes.
+
+If it returns `null` or `undefined`, Nuxt will fall back to the default routes (useful to modify input array).
+
+```js [app/router.options.ts]
+import type { RouterConfig } from '@nuxt/schema'
+
+// https://router.vuejs.org/api/interfaces/routeroptions.html
+export default <RouterConfig> {
+  routes: (_routes) => [
+    {
+      name: 'home',
+      path: '/',
+      component: () => import('~/pages/home.vue')
+    }
+  ],
+}
+```
+
+::alert
+Nuxt will not augment any new routes you return from the `routes` function with metadata defined in `definePageMeta` of the component you provide. If you want that to happen, you should use the `pages:extend` hook which is [called at build-time](/docs/api/advanced/hooks#nuxt-hooks-build-time).
+::
+
+### Using pages:extends hook
+
+You can add, change or remove pages from the scanned routes with the `pages:extend` nuxt hook. For example, to prevent creating routes for any `.ts` files:
```

Comment:
```suggestion
You can add, change or remove pages from the scanned routes with the `pages:extend` nuxt hook (or the `extendPages` utility which you can import from `@nuxt/kit`, if you are creating a module). For example, to prevent creating routes for any `.ts` files:
```